app/views/user_mailer/challenge_assignment_notification.html.erb

```diff
@@ -14,46 +14,46 @@
 
     <% def styled_tag_list(tags) %>
       <% return nil if !tags || tags.empty? %>
-      <% tags.map{|tag| style_link(tag.name, tag_works_url(tag))}.to_sentence.html_safe %>
+      <% tags.map { |tag| style_link(tag.name, tag_works_url(tag)) }.to_sentence.html_safe %>
     <% end %>
 
     <% fandoms = prompt.any_fandom ? "Any" : styled_tag_list(tag_groups["Fandom"]) %>
     <% chars = prompt.any_character ? "Any" : styled_tag_list(tag_groups["Character"]) %>
     <% ships = prompt.any_relationship ? "Any" : styled_tag_list(tag_groups["Relationship"]) %>
-    <% ratings = prompt.any_rating ? "Any" : get_title_string(tag_groups["Rating"]) %>
-    <% warnings = tag_groups["Warning"] ? get_title_string(tag_groups["Warning"]) : nil %>
-    <% categories = tag_groups["Category"] ? get_title_string(tag_groups["Category"]) : nil %>
-    <% atags = styled_tag_list(tag_groups["Freeform"]) %>
+    <% ratings = prompt.any_rating ? "Any" : (tag_groups["Rating"] ? get_title_string(tag_groups["Rating"]) : nil) %>
```

Comment:
Having multiple ? makes this hard to understand.

ratings = nil 
if prompt.any_rating then 
  ratings= "Any"
else 
  if tag_groups["Rating"]  then
    ratings = get_title_string(tag_groups["Rating"])
end